### PR TITLE
VDKANDROID-115: Add custom conditions to ControlSDK

### PR DESCRIPTION
**Pull request by indieboyalex**

### CODE REVIEW

File: control-sdk/src/androidTest/java/tv/accedo/one/sdk/AccedoOneTest.java [hunk 1]
```diff
@@ -152,6 +154,19 @@ public void testProfile() {
...
+    public void testProfileWithConditions() {
+        try {
+
+            Profile profile = accedoOneConditional.control().getProfile(Shared.getContext());
+            assertNotNull(profile);
+            assertNotNull(profile.getProfileId());
+        } catch (AccedoOneException e) {
+            e.printStackTrace();
+            fail();
+        }
```
Comment (ManuelPeinado) [inline control-sdk/src/androidTest/java/tv/accedo/one/sdk/AccedoOneTest.java:167]: This is not testing much at the moment right? We would have to add a new profile in Control that's only active when the required `cc` parameter is passed, and then check that the expected profile is returned (similar to what we're doing in `testValidGid` at the moment).

There's probably a good reason why you decided not to do that so feel free to ignore me, just figured I'd mention it.

Alternatively we could write the test in a different way, one that doesn't depend on the actual Control configuration that's deployed on the server:

```diff
    @Test
    public void testProfileWithConditions() {
        String finalUrl = accedoOneConditional.createRestClient("https://api.one.accedo.tv/profile").getUrl();
        HttpUrl parsedFinalUrl = HttpUrl.parse(finalUrl);
        Set<String> ccQueryParamValues = new HashSet<>(parsedFinalUrl.queryParameterValues("cc"));
        assertTrue(ccQueryParamValues.contains("mockKey1:mockValue1"));
        assertTrue(ccQueryParamValues.contains("mockKey2:mockValue2"));
    }
```

File: control-sdk/src/test/java/tv/accedo/one/SampleTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+public class SampleTest {
+
+@Test
+public void sampleTest() {
+    assert (true);
+}
+
+
+}
```
Comment (ManuelPeinado) [inline control-sdk/src/test/java/tv/accedo/one/SampleTest.java:13]: File perhaps added by accident?

**Review by ManuelPeinado — APPROVED** [review]
Looks great, just two nits.

File: control-sdk/src/androidTest/java/tv/accedo/one/sdk/AccedoOneTest.java (hunk 1 above)
Comment (indieboyalex) [reply control-sdk/src/androidTest/java/tv/accedo/one/sdk/AccedoOneTest.java:167]: In fact all I wanted that test to do was log the output so I could ensure the cc URL params had been added - which they were.

File: control-sdk/src/test/java/tv/accedo/one/SampleTest.java (hunk 2 above)
Comment (indieboyalex) [reply control-sdk/src/test/java/tv/accedo/one/SampleTest.java:13]: Shockingly no. But the CI PR check task kept failing without at least ONE file in the tests source path. So I created the smallest possible test I could just to get this working

**Pull request merged**